.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,117 @@
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # https://github.com/Azure/login?tab=readme-ov-file#login-with-openid-connect-oidc-recommended
+      - name: Azure Login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Download Analysis Outputs
+        id: download-analysis-outputs
+        uses: RMI-PACTA/actions/actions/azure/blob-copy@main
+        with:
+          source: https://pactadatadev.blob.core.windows.net/ghactions-workflow-pacta-results/main/latest/full_params_2023Q4/analysis_output_dir
+          destination: tmp_analysis_output_dir
+
+      - name: Download Benchmarks
+        id: download-benchmarks
+        uses: RMI-PACTA/actions/actions/azure/blob-copy@main
+        with:
+          source: https://pactadatadev.blob.core.windows.net/benchmarks-webapp/2023Q4/2023Q4_20240529T002355Z
+          destination: tmp_benchmarks_dir
+
+      - name: move output and analysis files to correct directory
+        run: |
+          mv tmp_analysis_output_dir/main/latest/full_params_2023Q4/analysis_output_dir analysis_output_dir
+          ls analysis_output_dir
+          mv tmp_benchmarks_dir/2023Q4/2023Q4_20240529T002355Z benchmarks_dir
+          ls benchmarks_dir
+
+
+      - name: Create Output Directories
+        run: |
+          mkdir -p dashboard_output_dir/data
+
+      - name: Run Docker Image
+        env:
+          FULL_IMAGE_NAME: ${{ inputs.full-image-name }}
+          WORKSPACE: ${{ github.workspace }}
+          PARAMETERS: ""
+          # PARAMETERS: ${{ steps.prepare.outputs.parameters }}
+          ANALYSIS_OUTPUT_DIR: "analysis_output_dir"
+          BENCHMARKS_DIR: "benchmarks_dir"
+          # PACTA_DATA_DIR: pacta-data
+          # PORTFOLIO_DIR: tests/portfolios
+          DASHBOARD_DATA_DIR: "dashboard_output_dir/data"
+          DASHBOARD_OUTPUT_DIR: "dashboard_output_dir"
+        run: |
+          chmod -R 777 dashboard_output_dir
+
+          docker run \
+          --network none \
+          --user $(id -u):$(id -g) \
+          --env LOG_LEVEL=TRACE \
+          --env ANALYSIS_OUTPUT_DIR="/mnt/analysis_output_dir" \
+          --env BENCHMARKS_DIR="/mnt/benchmarks_dir" \
+          --env DASHBOARD_DATA_DIR="/mnt/dashboard_output_dir/data" \
+          --env DASHBOARD_OUTPUT_DIR="/mnt/dashboard_output_dir" \
+          --mount type=bind,readonly,source=${WORKSPACE}/${BENCHMARKS_DIR},target=/mnt/benchmarks_dir \
+          --mount type=bind,source=${WORKSPACE}/${ANALYSIS_OUTPUT_DIR},target=/mnt/analysis_output_dir \
+          --mount type=bind,source=${WORKSPACE}/${DASHBOARD_OUTPUT_DIR},target=/mnt/dashboard_output_dir \
+          --mount type=bind,source=${WORKSPACE}/${DASHBOARD_DATA_DIR},target=/mnt/dashboard_output_dir/data \
+          $FULL_IMAGE_NAME \
+          "$PARAMETERS"
```

Comment:
In the spirit of keeping the `prod` and `dev` environments as closely aligned as possible, wouldn't it make more sense to just call the `docker-compose.yml` here? (otherwise, I'm not totally sure why we have a `docker-compose.yml` file)

https://12factor.net/dev-prod-parity

Reply:
The `docker-compose.yml` file is primarily a shortcut (to avoid having to type all the mounting and env flags). Additionally, that is (generally) set to `build: .`, rather than using a pre-built image (which is what we're testing here). 

We could add an override file with the CI settings, but passing the full image ID might start to get tricky there (unless we build an override file in the action).

the dev/prod parity issue isn't too much of a problem here, since we aren't using any backing services during execution, and all the backing services that we're using for environment setup (Azure Blobs) are inherently different than the ones we're using in PROD (Azure files, mounted to Container Instances), but we can't replicate that behavior on GHA.